amaranth/hdl/_dsl.py

```diff
@@ -16,7 +16,12 @@
 from ._mem import MemoryData
 
 
-__all__ = ["SyntaxError", "SyntaxWarning", "Module"]
+__all__ = ["FrozenModule", "SyntaxError", "SyntaxWarning", "Module"]
+
+
+@final
+class FrozenModule(Exception):
```

Comment:
I think "frozen" doesn't make sense as a designer-facing concept. Bikeshed:
- `ElaboratedModule`?
- `AlreadyElaborated`?
- something else?

Reply:
yeah, I was going for consistency with `FrozenMemory`

I like `AlreadyElaborated`, do you think we should use it here and also replace `FrozenMemory` with it?

Reply:
SGTM

Reply:
done; had to move the exception to `_ir` to avoid a circular import between `_dsl` and `_mem`